includes/Commerce/Orders.php

```diff
@@ -298,12 +299,82 @@ public function fulfill_order( \WC_Order $order, $tracking_number, $carrier ) {
 	 * @since 2.1.0-dev.1
 	 *
 	 * @param \WC_Order_Refund $refund order refund object
-	 * @param mixed $args
+	 * @param string $reason_code refund reason code
 	 * @throws SV_WC_Plugin_Exception
 	 */
-	public function add_order_refund( \WC_Order_Refund $refund, $args ) {
+	public function add_order_refund( \WC_Order_Refund $refund, $reason_code ) {
 
-		// TODO: implement
+		$plugin = facebook_for_woocommerce();
+
+		$api = $plugin->get_api( $plugin->get_connection_handler()->get_page_access_token() );
+
+		$valid_reason_codes = [
+			Refund_Request::REASON_BUYERS_REMORSE,
+			Refund_Request::REASON_DAMAGED_GOODS,
+			Refund_Request::REASON_NOT_AS_DESCRIBED,
+			Refund_Request::REASON_QUALITY_ISSUE,
+			Refund_Request::REASON_OTHER,
+			Refund_Request::REASON_WRONG_ITEM,
+		];
+
+		if ( ! in_array( $reason_code, $valid_reason_codes, true ) ) {
+			$reason_code = Refund_Request::REASON_OTHER;
+		}
+
+		try {
+
+			$parent_order = wc_get_order( $refund->get_parent_id() );
+
+			if ( ! $parent_order instanceof \WC_Order ) {
+				throw new SV_WC_Plugin_Exception( __( 'Parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$remote_id = $parent_order->get_meta( self::REMOTE_ID_META_KEY );
+
+			if ( ! $remote_id ) {
+				throw new SV_WC_Plugin_Exception( __( 'Remote ID for parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$items = [];
+
+			/** @var \WC_Order_Item_Product $item */
+			foreach ( $refund->get_items() as $item ) {
+
+				if ( $product = $item->get_product() ) {
+
+					$items[] = [
+						'retailer_id'          => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $product ),
+						'item_refund_quantity' => $item->get_quantity(),
```

Comment:
Refund item quantities are negative, so this will need to be converted to positive. I _think_ that's consistent across our minimum supported WC versions, but that needs to be confirmed to handle both cases if not.

Also, there's no guarantee that every line item will have a quantity. See the amount inputs in the refund UI - you can bump the quantity but also input amounts free-form. Looks like `item_refund_amount` could be used in that case.

Reply:
Do you mean always send the `item_refund_amount`, instead of the quantity?

Reply:
I was thinking, based on the doc descriptions, that if a quantity exists use that, then if not use the amount. What do you think?

Reply:
Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1525/commits/da235caba6d375faeed5e0d9fd4fd7aa126dfb12.